development/source-code-documentation.md

```diff
@@ -0,0 +1,187 @@
+# Documentation standards
+
+Code should be documented according to [JSDoc](https://jsdoc.app/) standards. Documentation comments should enable the
+TypeScript compatible [type checking](https://www.typescriptlang.org/docs/handbook/type-checking-javascript-files.html),
+if such type checking is supported by an editor or an IDE. For this we should follow the 
```

Comment:
can you make a recommendation for how to setup the IDEs that we use? (I use Atom, I'm not sure what you and Irina use)

Reply:
I use Visual Studio Code and I don't know how to set such a check too.

Reply:
I will try to add what I can although it might be second-hand because I use neither of those. But I think once we start using it more extensively, we'll get some first-hand experience and we can update documentation with it. I see these notes as a live document that would be updated every time we figure out something new and valuable.

Reply:
This is what I've found so far.

Atom can support type checks via Tern: https://atom.io/packages/atom-ternjs
VSCode: there is a built-in support for type checking. Here is an article that specifies how to use it there: https://medium.com/@trukrs/type-safe-javascript-with-jsdoc-7a2a63209b76.

Some more general information about static type checking is available [here](https://medium.com/javascript-scene/you-might-not-need-typescript-or-static-types-aa7cb670a77b).

I'll add it to the documentation.